ui/apps/platform/src/Containers/Audit/ListeningEndpoints/ListeningEndpointsTable.tsx

```diff
@@ -46,49 +46,81 @@ function EmbeddedTable({
 export type ListeningEndpointsTableProps = {
     deployments: (ListDeployment & { listeningEndpoints: ProcessListeningOnPort[] })[];
     getSortParams: GetSortParams;
+    areAllRowsExpanded: boolean;
+    setAllRowsExpanded: Dispatch<SetStateAction<boolean>>;
 };
 
-function ListeningEndpointsTable({ deployments, getSortParams }: ListeningEndpointsTableProps) {
-    const expandedRowSet = useSet<string>();
+function ListeningEndpointsTable({
+    deployments,
+    getSortParams,
+    areAllRowsExpanded,
+    setAllRowsExpanded,
+}: ListeningEndpointsTableProps) {
+    // This is used to track which table rows are in the -opposite- state
+    // of the passed expanded state for the entire table.
+    const invertedExpansionRowSet = useSet<string>();
+
     return (
         <TableComposable variant="compact" aria-label="Deployment results">
             <Thead noWrap>
                 <Tr>
-                    <Td width={10}>{/* Header for expanded column */}</Td>
+                    <Th
+                        expand={{
+                            // Possible PF bug? This boolean seems to need to be inverted based on the render output
+                            areAllExpanded: !areAllRowsExpanded,
+                            // TODO Awkward type assertion here is fixed in PF 5 https://github.com/patternfly/patternfly-react/issues/8330
+                            collapseAllAriaLabel: 'Expand or collapse all rows' as '',
+                            onToggle: () => {
+                                setAllRowsExpanded(!areAllRowsExpanded);
+                                invertedExpansionRowSet.clear();
+                            },
+                        }}
+                        width={10}
+                    >
+                        {/* Header for expanded column */}
```

Comment:
@dval can you see if axe DevTools reports issue? If yes, does `Td` trick mess up the behavior?

Reply:
No DevTools issue here. The `expand` prop causes the `Th` to have a button component with an `aria-label` prop, which provides suitable text content to screen readers.

(I checked when making this change, since the `Td` component does not accept the same `expand` prop as the `Th`. Fortunately this change didn't report an accessibility error.)

Reply:
Aha, React `Th` element appears as if empty, but HTML `th` element has content.